foo_uie_console/main.cpp

```diff
@@ -331,6 +331,11 @@ void ConsoleWindow::update_content()
         buffer << "[" << pfc::format_int(message.m_time.wHour, 2) << ":" << pfc::format_int(message.m_time.wMinute, 2)
                << ":" << pfc::format_int(message.m_time.wSecond, 2) << "] " << message.m_message.c_str();
     }
+
+    // Truncate the trailing newline
+    if (!buffer.is_empty())
+        buffer.truncate_last_char();
```

Comment:
A newline would normally be two characters here (CR/LF i.e. `\r\n`) so it'd be better to be a bit smarter here and truncate based on what the trailing characters are.

I think you could use an `std::string_view` and the [`find_last_not_of` method](https://en.cppreference.com/w/cpp/string/basic_string_view/find_last_not_of) for an easy way to work out where to truncate.